### PR TITLE
feat: add Amplify CLI command reference

### PR DESCRIPTION
**Pull request by josefaidt**
_Issue #, if available:_

#2110 
#5207 
#5292
#5311

_Description of changes:_

image

UPDATED

image

image

image

### Todo

- [x] ensure sidenav links are displayed as inline code blocks #4930  
- [ ] ~~ensure titles are displayed as the full command as an inline code block (e.g. `amplify env`)~~
- [x] ensure page SEO is working as expected
- [ ] ~~ensure "previous" and "next" buttons display commands as inline code blocks~~
- [ ] ~~remove ALL references to commands, their structure, and their usage, reference new pages instead~~
	- https://github.com/aws-amplify/docs/issues/5153
- [x] automate pulling commands from Amplify CLI

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by josefaidt** [comment]
adding do-not-merge to look into issue where data is getting passed around to command pages, where navigating away from and back to a specific command will carry subcommands, options, and flags back to the command page

File: package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "@emotion/styled": "^11.3.0",
     "@mdx-js/mdx": "^2.0.0",
     "algoliasearch": "^4.12.1",
+    "amplify-cli-core": "3.6.2",
```
Comment (edwardfoyle) [inline package.json:27]: should this be carat versioned?

File: src/data/cli-commands.mjs [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import { commandsInfo } from 'amplify-cli-core/lib/help/commands-info.js';
```
[the remark this answers is not in the document]
Comment (josefaidt) [reply src/data/cli-commands.mjs:1]: oh that's a good question. If this is changed in the CLI the docs site will not build. Ideally if we automate this on updates to `amplify-cli-core` the PR associated with the version bump and lockfile updates will fail

File: package.json (hunk 1 above)
Comment (josefaidt) [reply package.json:27]: I originally had this set to `latest`, which would only require a lockfile update when updating docs after a CLI release. It's pinned to be explicit, but I don't see any issues with adding a carat

Comment (sobolk) [reply package.json:27]: Also, please switch to `^4.0.0`

Comment (josefaidt) [reply package.json]: @sobolk I've bumped the version and changed the package name, though unfortunately I am unable to import the help information 
```ts
import { commandsInfo } from '@aws-amplify/amplify-cli-core';
```
image

Thankfully using yarn will allow us to still import the transient dependency, so it's not entirely necessary to update the import just yet

**Review by lazpavel — APPROVED** [review]

**Review by katiegoines — APPROVED** [review]

**Pull request merged**